File: src/java/iced-x86/src/test/java/com/github/icedland/iced/x86/CodeWriterImpl.java
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

package com.github.icedland.iced.x86;

import java.io.ByteArrayOutputStream;

public final class CodeWriterImpl implements CodeWriter {
	final ByteArrayOutputStream stream = new ByteArrayOutputStream();

	@Override
	public void writeByte(byte value) {
		stream.write(value);
	}

	public byte[] toArray() {
		return stream.toByteArray();
	}
}

File: src/csharp/Intel/Iced.UnitTests/Intel/CodeWriterImpl.cs
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

#if ENCODER
using System.Collections.Generic;
using Iced.Intel;

namespace Iced.UnitTests.Intel {
	sealed class CodeWriterImpl : CodeWriter {
		readonly List<byte> bytes = new List<byte>();
		public override void WriteByte(byte value) => bytes.Add(value);
		public byte[] ToArray() => bytes.ToArray();
	}
}
#endif
